suggest name when starting feature branch and convert spaces to underbars

jefmwols opened the pull request:
This changes the start command so that it suggests a branch-name suffix when asking for a branch name.  The suggestion is the name of the story with spaces replaced with underbars.  If the user does not type in a branch name, then the suggested suffix is added to the branch name. If the user does type in a branch name, then the string they typed is used.  If the resulting branch name contains spaces, they are replaced with underbars.

jefmwols committed:
suggest name when starting feature branch and convert spaces to underbars

jefmwols committed:
add ability to specify a feature_root branch

jefmwols committed:
move branch checking to its own method

jefmwols commented:
The last 2 commits add the ability to suggest and optionally move to a configures feature_root branch before beginning the start command.  The feature_root branch would be configured in --get git-pivotal-tracker-integration.feature-root.

jefmwols committed:
re[;ace all non-alphanumeric characters in branch name with underbar

jefmwols committed:
[#70304938]add ability to truncate branch name to at most 30 chars

jefmwols committed:
Merge 70304938_User_SBAT_see_the_auto_branc__ to master

[Completes #70304938]

jefmwols committed:
[Completes #70306124]

jefmwols committed:
modify gemfile

jefmwols committed:
Update README.md

jefmwols committed:
Merge branch 'master' of github.com:v2dev/V2GPTI

jefmwols committed:
add version-checking instructions to README [#70356510]

Commit:
Update README.md

Updated install section.

jefmwols committed:
[#70355978]remove special case for just one story

jefmwols committed:
Merge 70355978_System_should_NOT_automatica__ to master

[Completes #70355978]

jefmwols committed:
create install script

jefmwols committed:
Update README.md

jefmwols committed:
Update install

jefmwols committed:
remove install folder

jefmwols committed:
Update README.md

jefmwols committed:
Update README.md

jefmwols committed:
Update README.md

jefmwols committed:
Update README.md

jefmwols committed:
Update README.md

jonwolski commented on lib/git-pivotal-tracker-integration/command/start.rb:
```diff
@@ -47,12 +48,40 @@ def run(filter)
-    puts
-    branch_name
+      prefix = "#{story.id}-"
+      story_name = "#{story.name.gsub(/[^0-9a-z\\s]/i, '_')}"
```
This appears to be the source of the failing spec. `story` gets doubled (mocked?), so the unexpected call to `name` throws an error. You could add an expectation to the spec, but there might be a better way.

jefmwols committed:
Update .ruby-version

jonwolski committed:
Updates test doubles

Updates stubs and mocks so that test suite passes.
Corrects regex in feature branch name gsub so that the hyphen between
project numeric id and feature name is preserved.

jefmwols committed:
[#70436944]update gemspec and build gem

jefmwols committed:
Merge 70436944_Developer_SBAT_Gem_Install_f__ to master

[Completes #70436944]

jefmwols committed:
Update README.md

jefmwols committed:
Merge pull request #1 from jonwolski/feature/suggest-name-of-feature-branch

Updates test doubles
